### DISCUSSION TITLE
Prebuilt docker for 0.21

### DISCUSSION THREAD

**Discussion by donkirkby** [pyodide/pyodide#3304] [Q&A] [answered]
I'm trying to run a partial build, including my own package, so I tried to pull the latest prebuilt docker image and run the build.

The latest prebuilt docker image is version 0.20.0, but that's pointing at an old version of zlib that's no longer on their web site. `wget` downloads an empty file and the build fails.

```
$ git checkout tags/0.20.0
HEAD is now at 7a9273dd v0.20.0
$ sudo ./run_docker --pre-built PYODIDE_PACKAGES=core,space-tracer,matplotlib make
./tools/dependency-check.sh
date +"[%F %T] Building cpython..."
[2022-11-29 16:25:50] Building cpython...
make -C /src/cpython
make[1]: Entering directory '/src/cpython'
[ -d /src/cpython/build ] || (mkdir /src/cpython/build)
tar -C /src/cpython/build/ -xf /src/cpython/downloads/zlib-1.2.12.tar.gz
tar: This does not look like a tar archive

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
make[1]: *** [Makefile:90: /src/cpython/build/zlib-1.2.12/.configured] Error 2
make[1]: Leaving directory '/src/cpython'
make: *** [Makefile:219: /src/cpython/installs/python-3.10.2/lib/python3.10] Error 2
$ ls -l cpython/downloads/zlib-1.2.12.tar.gz 
-rw-r--r-- 1 root root 0 Nov 28 23:17 cpython/downloads/zlib-1.2.12.tar.gz
$ 
```

Are there plans to publish a prebuilt docker image for 0.21? My first attempt to build it myself failed, but I'll try digging into it.

For now, I downloaded the old version of zlib before running the build:

```
sudo wget -O cpython/downloads/zlib-1.2.12.tar.gz https://www.zlib.net/fossils/zlib-1.2.12.tar.gz
```

**Comment by ryanking13** [comment, marked as answer]
Thanks for the report @donkirkby. We have some issues in building pre-built docker images and so pre-built docker images for certain versions are missing.

By the way, you don't need the pre-built docker to build your package. It just helps re-building other packages faster, but it is not mandatory. You can use `./run_docker` without `--pre-built` flag.

> For now, I downloaded the old version of zlib before running the build:
> sudo wget -O cpython/downloads/zlib-1.2.12.tar.gz https://www.zlib.net/fossils/zlib-1.2.12.tar.gz

zlib 1.2.12 is yanked due to the vulnerability. In pyodide 0.20.0 branch, I guess you'll need the change the version in [`cpython/Makefile`](https://github.com/pyodide/pyodide/blob/7a9273dd6c893ab1273f8c5d22f62f60296f13d5/cpython/Makefile#L15)
